src/formatDate.js

```diff
@@ -8,7 +8,49 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formattedToDateParts = {}; // day, month, year
+  const values = date.split(fromFormat[3]);
+  let formattedDate = '';
+
+  for (let i = 0; i < 3; i++) {
+    if (fromFormat[i] === 'YYYY' && toFormat[i] === 'YY') {
+      formattedToDateParts[2] = values[i].slice(-2);
+    } else if (fromFormat[i] === 'YY' && toFormat[i] === 'YYYY') {
+      const year = Number(values[i]);
+
+      if (year < 30) {
+        formattedToDateParts[2] = '20' + values[i];
+      } else {
+        formattedToDateParts[2] = '19' + values[i];
```

Comment:
You are correctly converting `YY` to `YYYY`, but you are always assigning the result to `formattedToDateParts[2]`. This will cause an issue if the year is not meant to be in the third position in the `toFormat` array. You should assign the year to the correct position based on the `toFormat` array.